src/tools/crossgen2/Common/TypeSystem/Interop/IL/Marshaller.cs

```diff
@@ -1160,15 +1161,25 @@ protected override void TransformNativeToManaged(ILCodeStream codeStream)
             ILCodeLabel lRangeCheck = emitter.NewCodeLabel();
             ILCodeLabel lLoopHeader = emitter.NewCodeLabel();
             var lNullArray = emitter.NewCodeLabel();
-            
-            // Check for null array
-            LoadManagedValue(codeStream);
-            codeStream.Emit(ILOpcode.brfalse, lNullArray);
 
+            // Check for null array
+            if (!IsManagedByRef)
+            {
+                LoadManagedValue(codeStream);
+                codeStream.Emit(ILOpcode.brfalse, lNullArray);
+            }
 
             EmitElementCount(codeStream, MarshalDirection.Reverse);
 
             codeStream.EmitStLoc(vLength);
+
+            if (IsManagedByRef)
```

Comment:
Is the array byref common in any way? I would rather declare passing array byref to be unsupported by crossgen.

Reply:
I can just disable that part and we can re-enable if it ends up needed.